util/types.c

```diff
@@ -2,19 +2,23 @@
 
 #include <inttypes.h>
 #include <stdio.h>
+#include <string.h>
 
 #include "types.h"
 
-__uint128_t le128_to_cpu(__u8 *data)
+nvme_uint128_t le128_to_cpu(__u8 *data)
 {
-	int i;
-	__uint128_t result = 0;
+	nvme_uint128_t u;
 
-	for (i = 0; i < 16; i++) {
-		result *= 256;
-		result += data[15 - i];
-	}
-	return result;
+	memcpy(u.bytes, data, 16);
+
+#if HAVE_BIG_ENDIAN
```

Comment:
Forget it, I didn't realize the word ordering is also different.

Reply:
Hello,

looks like A am too late to the party.

The memcpy can be put in to the else branch.

And the #if can be changed to if() so that the BE branch gets compile-tested on LE.

Reply:
The memcpy is still needed to set the endian-reversed values. We could do this differently with a temporary variable, but since it's needed on LE anyway, may as well leave it there.